tests/containers/docker.pm

```diff
@@ -45,11 +46,14 @@ sub run {
     my $engine = $self->containers_factory('docker');
     test_seccomp();
 
-    if ($self->firewall() eq 'firewalld') {
-        # on publiccloud we need to install firewalld first
-        install_and_start_firewalld() if (is_publiccloud || is_jeos);
-        check_docker_firewall();
+    if ($self->firewall() == 'firewalld' && !is_microos()) {
```

Comment:
This check is a bit problematic. Because you would install `firewalld` only on publiccloud and jeos, but stop it on all flavors. I'd recommend to or move the `(is_publiccloud || is_jeos)` check to the upper `if` block or replace this check by a general check, if `firewalld` needs to be installed, e.g.

```suggestion
    if ($self->firewall() == 'firewalld' && !is_microos() && script_run('rpm -q firewalld') != 0) {
```

Otherwise we might end up at some point in a weird configuration.

Reply:
I also think this is the source of confusion for https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/13758#discussion_r760308575

Reply:
Indeed - we declare the variable to stop the firewall regardless of whether it’s the (IMHO also problematic) scenarios where we manually added a firewall

we should only stop the firewall on the systems that don’t normally have one, and we should probably uninstall it also, and we should probably do all that only in addition to testing the default configuration

Reply:
Please can you re-check?